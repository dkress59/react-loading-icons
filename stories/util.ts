import { ArgTypes } from '@storybook/react'
import { SVGProps } from 'react'

export type IconProps = JSX.IntrinsicAttributes & SVGProps<SVGSVGElement>
export const IconFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: 'transparent',
	strokeOpacity: 1,
	width: 'auto',
}
export const IconStroke: IconProps = {
	fill: 'transparent',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}
export const IconStrokeFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}

export const IconFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: 'transparent',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: 'transparent',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
