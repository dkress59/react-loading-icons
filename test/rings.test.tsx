import { render, screen } from '@testing-library/react'
import React from 'react'

import Rings from '../src/components/rings'
import { testProps } from './util'

describe('<Rings /> component', () => {
	it('renders and matches snapshot', () => {
		render(<Rings data-testid="icon" />)
		const svg = screen.getByTestId('icon')
		expect(svg).not.toBeNull()
		expect(svg).toMatchSnapshot()
		expect(svg.getAttributeNames()).toContain('viewBox')
		expect(svg.getAttribute('class')).toBe('icon-loading')
	})
	it('inherits props', () => {
		render(<Rings {...testProps} />)
		const svg = screen.getByTestId('icon')
		expect(svg.getAttribute('height')).toBe('2em')
		expect(svg.getAttribute('width')).toBe('auto')
		expect(svg.getAttribute('class')).toBe('icon-loading test')
	})
})
