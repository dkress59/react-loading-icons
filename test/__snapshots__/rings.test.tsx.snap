// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rings /> component renders and matches snapshot 1`] = `
<svg
  class="icon-loading"
  data-testid="icon"
  height="45"
  viewBox="0 0 45 45"
  width="45"
>
  <g
    fill="none"
    fill-rule="evenodd"
    stroke="#fff"
    stroke-width="2"
    transform="translate(1 1)"
  >
    <circle
      cx="22"
      cy="22"
      r="6"
    >
      <animate
        attributeName="r"
        begin="1.5s"
        calcMode="linear"
        dur="3s"
        repeatCount="indefinite"
        values="6;22"
      />
      <animate
        attributeName="stroke-opacity"
        begin="1.5s"
        calcMode="linear"
        dur="3s"
        repeatCount="indefinite"
        values="1;0"
      />
      <animate
        attributeName="stroke-width"
        begin="1.5s"
        calcMode="linear"
        dur="3s"
        repeatCount="indefinite"
        values="2;0"
      />
    </circle>
    <circle
      cx="22"
      cy="22"
      r="6"
    >
      <animate
        attributeName="r"
        begin="3s"
        calcMode="linear"
        dur="3s"
        repeatCount="indefinite"
        values="6;22"
      />
      <animate
        attributeName="stroke-opacity"
        begin="3s"
        calcMode="linear"
        dur="3s"
        repeatCount="indefinite"
        values="1;0"
      />
      <animate
        attributeName="stroke-width"
        begin="3s"
        calcMode="linear"
        dur="3s"
        repeatCount="indefinite"
        values="2;0"
      />
    </circle>
    <circle
      cx="22"
      cy="22"
      r="8"
    >
      <animate
        attributeName="r"
        begin="0s"
        calcMode="linear"
        dur="1.5s"
        repeatCount="indefinite"
        values="6;1;2;3;4;5;6"
      />
    </circle>
  </g>
</svg>
`;
